eZ/Publish/Core/Persistence/Legacy/Tests/User/UserHandlerTest.php

```diff
@@ -144,6 +154,73 @@ public function testLoadUserByEmail()
         );
     }
 
+    /**
+     * @expectedException \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function testLoadUserByTokenNotFound()
+    {
+        $handler = $this->getUserHandler();
+        $handler->create($user = $this->getValidUser());
+        $handler->updateUserToken($this->getValidUserToken());
+
+        $handler->loadUserByToken('asd');
+    }
+
+    public function testLoadUserByToken()
+    {
+        $handler = $this->getUserHandler();
+        $handler->create($user = $this->getValidUser());
+        $handler->updateUserToken($userToken = $this->getValidUserToken());
+
+        $loadedUser = $handler->loadUserByToken($userToken->hashKey);
+        $this->assertEquals(
+            $user,
+            $loadedUser
+        );
+    }
+
+    public function testUpdateUserToken()
+    {
+        $handler = $this->getUserHandler();
+
+        $handler->updateUserToken($userToken = $this->getValidUserToken(1234567890));
+
+        $this->assertQueryResult(
+            [['0800fc577294c34e0b28ad2839435945', 1, 1234567890, 42]],
+            $this->handler->createSelectQuery()->select('*')->from('ezuser_accountkey'),
+            'Expected user data to be updated.'
+        );
+
+        $handler->updateUserToken($userToken = $this->getValidUserToken(2234567890));
+
+        $this->assertQueryResult(
+            [['0800fc577294c34e0b28ad2839435945', 1, 2234567890, 42]],
+            $this->handler->createSelectQuery()->select('*')->from('ezuser_accountkey'),
```

Comment:
Hmm, unit tests should not use real database, they do in some places by accident. I tried to do some integration on gateway level, but the code there also will need to be refactored.

What we need instead of this are tests in `\eZ\Publish\API\Repository\Tests\UserServiceTest`.
The more practical reason for that is that is that unit tests if use DB by accident, it is Sqlite only. Integration tests do it for MySQL, PostgreSQL and Sqlite.